x-pack/legacy/plugins/siem/public/components/links/index.test.tsx

```diff
@@ -103,51 +117,89 @@ describe('Custom Links', () => {
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(0)
+          .childAt(0)
+          .prop('href')
+      ).toEqual('https://www.virustotal.com/gui/search/192.0.2.0');
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(1)
+          .childAt(0)
+          .prop('href')
+      ).toEqual('https://talosintelligence.com/reputation_center/lookup?search=192.0.2.0');
     });
-  });
 
-  describe('VirusTotalLink', () => {
-    test('it renders sha passed in as value', () => {
-      const wrapper = mountWithIntl(<VirusTotalLink link={'abc'}>{'Example Link'}</VirusTotalLink>);
-      expect(wrapper.text()).toEqual('Example Link');
-    });
+    test("it encodes <script>alert('XSS')</script>", () => {
+      const wrapper = mountWithIntl(<ReputationLink domain={"<script>alert('XSS')</script>"} />);
 
-    test('it renders sha passed in as link', () => {
-      const wrapper = mountWithIntl(
-        <VirusTotalLink link={'abc'}>{'Example Link'} </VirusTotalLink>
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(0)
+          .childAt(0)
+          .prop('href')
+      ).toEqual("https://www.virustotal.com/gui/search/%3Cscript%3Ealert('XSS')%3C%2Fscript%3E");
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(1)
+          .childAt(0)
+          .prop('href')
+      ).toEqual(
+        "https://talosintelligence.com/reputation_center/lookup?search=%3Cscript%3Ealert('XSS')%3C%2Fscript%3E"
       );
-      expect(wrapper.find('a').prop('href')).toEqual('https://www.virustotal.com/#/search/abc');
     });
 
-    test("it encodes <script>alert('XSS')</script>", () => {
-      const wrapper = mountWithIntl(
-        <VirusTotalLink link={"<script>alert('XSS')</script>"}>{'Example Link'}</VirusTotalLink>
-      );
-      expect(wrapper.find('a').prop('href')).toEqual(
-        "https://www.virustotal.com/#/search/%3Cscript%3Ealert('XSS')%3C%2Fscript%3E"
-      );
+    test('it renders correct number of links', () => {
+      (useUiSetting$ as jest.Mock).mockReset();
+      (useUiSetting$ as jest.Mock).mockReturnValue([
+        [
+          { name: 'Link 1', url_template: 'https://www.virustotal.com/gui/search/{{ip}}' },
+          {
+            name: 'Link 2',
+            url_template: 'https://talosintelligence.com/reputation_center/lookup?search={{ip}}',
+          },
+          { name: 'Link 3', url_template: 'https://www.virustotal.com/gui/search/{{ip}}' },
+          {
+            name: 'Link 4',
+            url_template: 'https://talosintelligence.com/reputation_center/lookup?search={{ip}}',
+          },
+          { name: 'Link 5', url_template: 'https://www.virustotal.com/gui/search/{{ip}}' },
+          {
+            name: 'Link 6',
+            url_template: 'https://talosintelligence.com/reputation_center/lookup?search={{ip}}',
+          },
+        ],
+      ]);
+      let wrapper = mountWithIntl(<ReputationLink domain={'192.0.2.0'} />);
```

Comment:
If this test requires mutating the wrapper and multiple expects, consider refactoring into multiple tests, where each test has one `expect`